### ISSUE TITLE
Inconsistent length of dashes at the same way

### ISSUE THREAD

**Issue by Helium314** [tangrams/tangram-es#2260]
**TO REPRODUCE THE ISSUE, FOLLOW THESE STEPS:**

 - Have StreetComplete 31.2 (probably other versions work as well)
 - go to [this location](https://www.openstreetmap.org/#map=19/48.31856/16.32992)

**RESULT:**

Dashes of one part of the track change length depending on zoom level
![dashes](https://user-images.githubusercontent.com/43007630/115008538-843b3100-9eab-11eb-8ca3-4ef77a43bdf4.png)

**EXPECTED RESULT:**

Consistent length of dashes over the whole track, as it is the case for (almost?) all other ways with dashed outlines.

**ENVIRONMENT:**

 - Android 9, Lineage 16.0, S4 Mini
 - StreetComplete 31.2

**OTHER:**

This is the only place where I found this issue after searching for a while, so I guess it's pretty uncommon.

original issue: https://github.com/streetcomplete/streetcomplete-mapstyle/issues/95

**Comment by matteblair** [comment]
Thanks for the issue report! I confirmed that this incorrect rendering is caused by Tangram ES and I've implemented a fix. This effect is caused by "overzooming" a tile with line dashes, i.e. zooming in to a higher zoom level than the source tile data. In the streetcomplete scene file, the tile source has a `max_zoom` of 16. When you zoom in to zoom level 20, you are actually looking at a zoom 16 tile "overzoomed" by 4 zoom levels, or by a factor of 16. This caused numerical precision issues in the UV coordinates used to render line dashes (among other things). The fix for this is simple: I changed the scale value used to pack polyline UV coordinates such that a greater range of UV values can be stored accurately. This reduces some amount of precision in polyline UV coordinates, which _could_ be noticeable if a texture larger than 2048px were used with a polyline, but this seems unlikely in practice.
